jina/peapods/pea.py

```diff
@@ -360,35 +339,49 @@ def load_plugins(self):
             from ..importer import PathImporter
             PathImporter.add_modules(*self.args.py_modules)
 
-    def loop_teardown(self):
-        """Stop the request loop """
+    def close_zmqlet(self):
+        """Close the zmqlet if exists"""
         if hasattr(self, 'zmqlet'):
             self.zmqlet.close()
 
-    def run(self):
-        """Start the request loop of this BasePea. It will listen to the network protobuf message via ZeroMQ. """
+    def _initialize_executor(self):
         try:
-            # Every logger created in this process will be identified by the `Pod Id` and use the same name
-            self.post_init()
-            self.loop_body()
+            os.environ['JINA_POD_NAME'] = self.name
+            os.environ['JINA_LOG_ID'] = self.args.log_id
+            self.load_plugins()
+            self.load_executor()
+            return self.executor
         except ExecutorFailToLoad:
             self.logger.critical(f'can not start a executor from {self.args.uses}', exc_info=True)
-        except KeyboardInterrupt:
```

Comment:
I don’t see the reason of removing them here and then trust devs to catch them

Reply:
Keyboard interrupt is caught and a log message is shown. do not understand the concern